src/ralph/api/auth/user.py

```diff
@@ -18,6 +19,42 @@
 ]
 
 
+class UserScopes(FrozenSet[Scope]):
+    """Scopes available to users."""
+
+    @lru_cache()
+    def is_authorized(self, requested_scope: Scope):
+        """Check if the requested scope can be accessed based on user scopes."""
+        expanded_scopes = {
+            "statements/read": {"statements/read/mine", "statements/read"},
+            "all/read": {
+                "statements/read/mine",
+                "statements/read",
+                "state/read",
+                "profile/read",
+                "all/read",
+            },
+            "all": {
+                "statements/write",
+                "statements/read/mine",
+                "statements/read",
+                "state/read",
+                "state/write",
+                "define",
+                "profile/read",
+                "profile/write",
+                "all/read",
+                "all",
+            },
+        }
```

Comment:
> Wouldn't it be possible to define them with a wildcard? E.g. statements/read/* and */read/*?
(initially commented by @wilbrdt)

I prefered this explicit syntax as it seemed clearer when reading the code, but I can change this. Anyone else has an opinion ?

Reply:
I'm fine with this explicit syntax!

Reply:
Mybe `Scope` and expanded scopes can be defined as `Enum`s listing all possible (expanded) scopes. This will be easier to manipulate if we should refer to a particular (expanded) scope. WDYT?

Reply:
To be changed later : https://github.com/openfun/ralph/issues/479